cloud: add dataservice path parameters specification (#17491)
pingcap/docs: tidb-cloud/data-service-manage-endpoint.md

- **Path**: the path that users use to access the endpoint.

    - The length of the path must be less than 64 characters.
    - The combination of the request method and the path must be unique within a Data App.
    - Usually, only letters, numbers, underscores (`_`), and slashes (`/`) are allowed in a path. A path must start with a slash (`/`) and end with a letter, number, or underscore (`_`). For example, `/my_endpoint/get_id`.
    - Path supports dynamic parameters, which need to be wrapped in curly braces, such as `{var}`. The name of the path parameter supports letters, numbers, and underscores, and can only start with a letter or an underline.

    > **Note:**
    >

>     > - The path parameter needs to occupy a separate level and does not support prefixes and suffixes.

Reviewer: ```suggestion
    > - In a path, each parameter must be at a separate level and do not support prefixes or suffixes.
```

    >

>     >    *Valid*: ```/var/{var}、/{var}```
>     >
>     >    *Invalid*:```/var{var}、/{var}var```

Reviewer: ```suggestion
    >    *Valid path*: ```/var/{var}``` and  ```/{var}```
    >
    >    *Invalid path*:```/var{var}``` and ```/{var}var```
```

    >

>     > - Paths with the same method and prefix may cause path conflicts, as shown in the following example:.

Reviewer: ```suggestion
    > - Paths with the same method and prefix might conflict, as in the following example:
```

    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/{var2}```
    >

>     >   These two path will conflict with each other because `GET /var/123` matches both;

Reviewer: ```suggestion
    >   These two paths will conflict with each other because `GET /var/123` matches both.
    >     
```

>     > - Paths with dynamic parameters have lower priority than absolute paths, such as:

Reviewer: ```suggestion
    > - Paths with parameters have lower priority than paths without parameters. For example:
```

    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/123```
    >

>     >   These two routes will not conflict because `GET /var/123` prioritizes matching with the latter;

Reviewer: ```suggestion
    >   These two paths will not conflict because `GET /var/123` takes precedence.
```

>     > - The Path parameter can be directly used in SQL, and more information can be found in [Configure parameters](#configure-parameters)

Reviewer: ```suggestion
    > - Path parameters can be used directly in SQL. For more information, see [Configure parameters](#configure-parameters).
```

- **Path**: the path that users use to access the endpoint.

    - The length of the path must be less than 64 characters.
    - The combination of the request method and the path must be unique within a Data App.
    - Only letters, numbers, underscores (`_`), slashes (`/`), and parameters enclosed in curly braces (such as `{var}`) are allowed in a path. Each path must start with a slash (`/`) and end with a letter, number, or underscore (`_`). For example, `/my_endpoint/get_id`.
    - For parameters enclosed in `{ }`, only letters, numbers, and underscores (`_`) are allowed. Each dynamic parameter must start with a letter or (`_`). 

    > **Note:**
    >

>     > - In a path, each parameter must be at a separate level and do not support prefixes or suffixes.

Reviewer: ```suggestion
    > - In a path, each parameter must be at a separate level and does not support prefixes or suffixes.
```

    >
    >    *Valid path*: ```/var/{var}``` and  ```/{var}```
    >

>     >    *Invalid path*:```/var{var}``` and ```/{var}var```

Reviewer 2: ```suggestion
    >    *Invalid path*: ```/var{var}``` and ```/{var}var```
```

    >
    > - Paths with the same method and prefix might conflict, as in the following example:
    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/{var2}```
    >
    >   These two paths will conflict with each other because `GET /var/123` matches both.
    >     
    > - Paths with parameters have lower priority than paths without parameters. For example:
    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/123```
    >

>     >   These two paths will not conflict because `GET /var/123` takes precedence.
>     > - Path parameters can be used directly in SQL. For more information, see [Configure parameters](#configure-parameters).

Reviewer 2: ```suggestion
    >   These two paths will not conflict because `GET /var/123` takes precedence.
    >
    > - Path parameters can be used directly in SQL. For more information, see [Configure parameters](#configure-parameters).
```

    - For parameters enclosed in `{ }`, only letters, numbers, and underscores (`_`) are allowed. Each dynamic parameter must start with a letter or (`_`). 

    > **Note:**
    >
    > - In a path, each parameter must be at a separate level and does not support prefixes or suffixes.
    >
    >    *Valid path*: ```/var/{var}``` and  ```/{var}```
    >
    >    *Invalid path*: ```/var{var}``` and ```/{var}var```
    >
    > - Paths with the same method and prefix might conflict, as in the following example:
    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/{var2}```
    >
    >   These two paths will conflict with each other because `GET /var/123` matches both.
    >     
    > - Paths with parameters have lower priority than paths without parameters. For example:
    >
    >    ```GET /var/{var1}```
    >
    >    ```GET /var/123```
    >
    >   These two paths will not conflict because `GET /var/123` takes precedence.
    >
    > - Path parameters can be used directly in SQL. For more information, see [Configure parameters](#configure-parameters).

> 

Reviewer 3: ```suggestion
```

- The parameter name: the name can only include letters, digits, and underscores (`_`) and must start with a letter or an underscore (`_`). **DO NOT** use `page` and `page_size` as parameter names, which are reserved for pagination of request results.

> - **Required**: specifies whether the parameter is required in the request. The default configuration is set to not required.
>     > **Tip:**
>     >
>     > The path parameter is required by default and cannot be modified.

Reviewer: ```suggestion
- **Required**: specifies whether the parameter is required in the request. For query parameters, the default configuration is not required. For path parameters, the configuration is required and cannot be modified.
```

> - **Type**: specifies the data type of the parameter. Supported values are `STRING`, `NUMBER`, `INTEGER`, `BOOLEAN`, and `ARRAY`. When using a `STRING` type parameter, you do not need to add quotation marks (`'` or `"`). For example, `foo` is valid for the `STRING` type and is processed as `"foo"`, whereas `"foo"` is processed as `"\"foo\""`.
>     > **Tip:**
>     >
>     > The path parameter currently only supports two types: `STRING` and `INTEGER`.

Reviewer: ```suggestion
- **Type**: specifies the data type of the parameter. For query parameters, `STRING`, `NUMBER`, `INTEGER`, `BOOLEAN`, and `ARRAY` are supported. For path parameters, only `STRING` and `INTEGER` are supported. 

    When using a `STRING` type parameter, you do not need to add quotation marks (`'` or `"`). For example, `foo` is valid for the `STRING` type and is processed as `"foo"`, whereas `"foo"` is processed as `"\"foo\""`.

```

    - For `ARRAY` type, you need to separate multiple values with a comma (`,`).
    - Make sure that the value can be converted to the type of parameter. Otherwise, the endpoint returns an error.
    - If you do not set a test value for a parameter, the default value is used when testing the endpoint.

> - **Location**: specifies the location of the parameter and can not be modified.
>     - For path parameters, the location is `Path`.
>     - For others, if the request method is `GET` or `DELETE`, parameter's location is `Query`, if it's `POST` or `PUT`, locatioin is `Body`.

Reviewer: ```suggestion
- **Location**: indicates the location of the parameter. This property cannot be modified.
    - For query parameters, if the request method is `GET` or `DELETE`, this property is `Query`. If the request method is `POST` or `PUT`, this property is `Body`.
    - For path parameters, this property is `Path`.
```

Author: 不止有path参数和query参数，还有body中的参数，所有的method都支持path参数，另外GET、DELETE支持query参数,  POST、PUT 支持body参数
